### PR TITLE
2893 - prefilled data total primary forest by climatic domaintab 1b

### LINKED ISSUE #2893: Prefilled data- Total -  Primary forest by climatic domain(tab 1b)
@minotogna 

The data source for the **Total** of the _Primary forest by climatic domain_ is from FRA 2020, explaining why 2025 data is missing. Not sure why the data source is not from the first table-FRA 2025.

One more concern is that the validation check is activated and highlights the **Total** even though there is no data entry.

### PR DESCRIPTION
**Pull request by minotogna**
https://github.com/openforis/fra-platform/assets/1138156/03feda8d-6ca7-4cdc-8dd6-6b516e7852b5

https://github.com/openforis/fra-platform/assets/1138156/110cca40-dc6f-4109-acf9-5c3fd49a5ba7

https://github.com/openforis/fra-platform/assets/1138156/ccb61bc9-b539-4daa-b73a-2cd6f8655d84

### CODE REVIEW

File: src/meta/assessment/nodeCalculations.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  data: RecordCountryData
+}
+
+const calculateIf = (props: Props): boolean => {
+  const { assessment, cycle, countryIso, row, col, data } = props
+  const { colName } = col.props
+  const formula = row.props.calculateIf?.[cycle.uuid]
+
+  const paramsCalculate = { assessment, countryIso, cycle, data, colName, row, formula }
+  return Boolean(ExpressionEvaluator.evalFormula<boolean>(paramsCalculate))
```
Comment (sorja) [inline src/meta/assessment/nodeCalculations.ts:30]: is this called twice? if so, can we avoid?

File: src/meta/assessment/nodeCalculations.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const value = RecordAssessmentDatas.getNodeValue(paramsValue)
+
+  // verify node value has not been inserted manually (see mirror tables) || calculateIf is verified
+  const canCalculate = row.props.calculateIf?.[cycle.uuid]
+    ? calculateIf(props)
+    : Objects.isEmpty(value) || value.calculated
+
+  if (canCalculate) {
+    const paramsCalculate = { assessment, countryIso, cycle, data, colName, row, formula }
+    const rawResult = ExpressionEvaluator.evalFormula<string | undefined>(paramsCalculate)
```
Comment (sorja) [inline src/meta/assessment/nodeCalculations.ts:56]: seems that the `rawResult` is recalculated if canCalculate has a true value

File: src/meta/assessment/nodeCalculations.ts (hunk 1 above)
Comment (minotogna) [reply src/meta/assessment/nodeCalculations.ts:30]: what is called twice?

File: src/meta/assessment/nodeCalculations.ts (hunk 2 above)
Comment (minotogna) [reply src/meta/assessment/nodeCalculations.ts:56]: exactly . it is meant that.

Comment (minotogna) [reply src/meta/assessment/nodeCalculations.ts:56]: oh , maybe you're confusing the calcualteIf vs calculateFn ? 
calculateIf is a separate function.

**Review by sorja — APPROVED** [review]

**Pull request merged**